docs/DeveloperGuide.md

```diff
@@ -1,38 +1,433 @@
+will reduce the chance of having an error if the user tries to select a time slot of the module that is not offered in
+the current semester.
+
+##### Alternatives considered.
+We thought of displaying the full module details from the search results. However, we decided against it as it would be
+too tedious for the user to search for the **exact module code** first before getting the details. The user may
+not know the exact module code, which is not very user-friendly and takes up a lot of time just to get the module
+details for 1 module.
+
+#### 3.4.8 ViewCommand
+
+The <code>ViewCommand</code> class extends from the <code>Command</code> class and displays the timetable of the current state's semester
+selected modules.
+
+#### 3.4.9 ExitCommand
+The <code>ExitCommand</code> class extends from the <code>Command</code> class and exits the program.
+
+### 3.5 Utils Component
+
+#### 3.5.1 UI Component
+
+![UI Class](images/Ui.png)
+
+The <code>UI</code> component can:
+- read input from the user
+- store and print out messages to the user
+
+##### Why it is implemented this way
+To comply with the Model-View Controller Framework
+To separate the internal representations and processing of information from the presentation and acceptance of
+information from the user
+
+##### Alternative Considered
+Each component to handle the presentation of information to the user
+
+- Increase coupling between components
+- Hard to define the sequence that information being presented to user
+
+#### 3.5.2 Link Component
+
+![Link Class](images/linkClass.png)
+
+The <code>Link</code> component can:
+- create a [NUSmod](https://nusmods.com/) link to be used in a browser
+- Parse a NUSmod link to import modules into YAMOM
+
+##### Why is it implemented this way
+To separate out the handling of NUSmod compatibility
+
+##### Alternative Considered
+To implement the handling of export in Storage class and import in Command class
+
+- Increase coupling between components
+- Decrease reusability of NUSmod link processing
+
+#### 3.5.3 Storage Component
+
+![Storage Class](images/storageClass.png)
+
+The <code>Storage</code> component can:
+
+- read from the hard disk a single line which is supposed to be a NUSMods export link
+- save to the hard disk
+
+Different checks have been implemented to ensure that even
+if the data file is modified in any way, it would not crash the programme.
+
+##### Why it is implemented this way
+
+To facilitate easy transfer of information from NUSMods to YAMOM.
+
+##### Alternatives considered
+
+Storing as <code>.json</code> file
+
+- would not be readable by the user
+- would have to implement another function for export/import function
+
+## 4. Implementation
+
+This section describes how key features of YAMOM are implemented in the latest release V1.0
+
+### Storage feature
+
+!["Opening saved state"](images/storageOpenPreviousState.png)  
```

Comment:
<img width="386" alt="pr1" src="https://user-images.githubusercontent.com/116053971/198250572-6f41bddf-decc-46ec-9087-74fe2ccc195a.png">

Are these activation bars supposed to be left open? Are they supposed to have return arrows?